geonode/static/geonode/js/upload/LayerInfo.js

```diff
@@ -477,6 +501,106 @@ define(function (require, exports) {
         }
     };
 
+    LayerInfo.prototype.sendData = function(data, url, done){
+
+        $.ajax({
+            url: url,
+            type: "POST",
+            data: data,
+            cache: false,
+            contentType: false,
+            processData: false
+        }).done(function(data){
+            done(data);
+        });
+
+    }
+
+    LayerInfo.prototype._chunkUpload = function(file, url, uuid, onSuccess) {
+        var loaded = 0;
+        var step = 1024; // Size of a chunk
+        var total = file.size;
+        var start = 0;
+        var stepCount = 0;
+        var reader = new FileReader();
+        var blob = file.slice(start,step);// Read each chunk
+        reader.readAsArrayBuffer(blob);
+        var self = this;
+
+        // this function will be invoked each time a file reader read a file
+        // Thus will be recursively called for each chunk
+        reader.onload = function(e){
+            console.log("Step Count: " + stepCount++ + " for file: " + file.name);
+            //create a blob from reader
+            var blob =  new Blob([reader.result]);
+            //create form data to send to server
+            var formData = new FormData();
+            formData.append("file", blob);
+            formData.append("filename", file.name);
+            formData.append("offset", loaded);
+            formData.append("file_size", total);
+
+            //send it via ajax
+            self.sendData(formData, url + "/" + uuid, function(data){
+                //if ajax done, do it recursively
+                //we already upload another step of file, add it to total uploaded
+                loaded += step;
+                if(loaded < total){
+                    //read again the file
+                    blob = file.slice(loaded, loaded + step);
+                    reader.readAsArrayBuffer(blob);
+                    //after read the blob, reader.onload will be revoked again
+                }else{
+                    //file is uploaded completely
+                    loaded = total; // just to stop the loop
+                    //call the callback function
+                    onSuccess(data);
+                }
+
+            });
+        }
+    }
+
+    function guid() {
+        return s4() + s4() + '-' + s4() + '-' + s4() + '-' +
+            s4() + '-' + s4() + s4() + s4();
+    }
+
+    function s4() {
+        return Math.floor((1 + Math.random()) * 0x10000)
+            .toString(16)
+            .substring(1);
+    }
+
+    LayerInfo.prototype.chunkUpload = function(index, uuid) {
+        //get all files that listed with this layer
+        var files = this.files;
+        var self = this;
+        var url = "/layers/api/chunk-file-uploader";
```

Comment:
Can this url becomes parameter, so you can reverse it in the template? Just to make sure if someone changes urls, it doesn't break.